tutorials/cifar10-distributed.ipynb

```diff
@@ -0,0 +1,1153 @@
+        "    checkpoint = torch.load(checkpoint_fp.as_posix(), map_location=\"cpu\")\n",
+        "    return checkpoint"
+      ],
+      "execution_count": 53,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "aBd6WDVE_KmO"
+      },
+      "source": [
+        "### Create Trainer\n",
+        "\n",
+        "Finally, we can create our `trainer` in four steps:\n",
+        "1. Choose whether to enable [Automatic Mixed Precision](https://pytorch.org/docs/stable/amp.html) (AMP) or not. Enabling AMP will speed up computations on large neural networks and reduce memory usage while retaining performance. A [`GradScaler()`](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler) object will be created to scale the `loss` so that gradients don't round-up to zero while training. \n",
+        "2. Define steps taken to process a single batch (`train_step()`)\n",
+        "  1. Move the batch to `device` used in current distributed configuration.\n",
+        "  2. Put `model` in `train()` mode.\n",
+        "  3. Perform forward pass by passing the inputs (`x`) through the `model` and calculating `loss`. If AMP is enabled then this step happens with [`autocast`](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.autocast) on which allows this step to run in mixed precision.\n",
+        "  4. Perform backward pass. If AMP is enabled, then the losses will be [scaled](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.scale) before calling `backward()`, `step()` the optimizer while discarding batches that contain NaNs and [update()](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.update) the scale for the next iteration.\n",
+        "3. Setup some common Ignite training handlers. You can do this individually or use [setup_common_training_handlers()](https://pytorch.org/ignite/contrib/engines.html#ignite.contrib.engines.common.setup_common_training_handlers) that takes the `trainer` and a subset of the dataset (`train_sampler`) alongwith:\n",
+        "  * A dictionary mapping on what to save in the checkpoint (`to_save`) and how often (`save_every_iters`).\n",
+        "  * The LR Scheduler\n",
+        "  * The output of `train_step()`\n",
+        "  * Other handlers\n",
+        "4. If `resume_from` file path is provided, load the states of objects `to_save` from the checkpoint file."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ptmfSvESbEPE"
+      },
+      "source": [
+        "def create_trainer(\n",
+        "    model, optimizer, criterion, lr_scheduler, train_sampler, config, logger\n",
+        "):\n",
+        "\n",
+        "    device = idist.device()\n",
+        "\n",
+        "    with_amp = config[\"with_amp\"]\n",
+        "    scaler = GradScaler(enabled=with_amp)\n",
+        "\n",
+        "    def train_step(engine, batch):\n",
```

Comment:
Why not use the `create_supervised_trainer` helper ? It handles `amp`, `apex` and `tpu`.

https://github.com/pytorch/ignite/blob/31738d47716d3055c356f29b9e01e022062041ea/ignite/engine/__init__.py#L318

Reply:
I wasn't aware about that, I'll add it at once!